source/base/subscriptor.cc

```diff
@@ -51,6 +51,7 @@ Subscriptor::Subscriptor(Subscriptor &&subscriptor) noexcept
   : counter(0)
   , object_info(subscriptor.object_info)
 {
+  std::lock_guard<std::mutex> lock(mutex);
   for (auto validity_ptr : subscriptor.validity_pointers)
     *validity_ptr = false;
```

Comment:
Hm, this only works because `mutex` is a `static` member (i.e., global) object. That seems heavy handed, though in practice probably not terrible.

We should probably merge this as is, but I wonder whether we can come up with a more fine-grained approach of using a non-`static` mutex?

Reply:
I just update the PR to use `std::atomic<bool>` for the pointers.